app/assets/javascripts/admin/controllers/admin-user-index.js.es6

```diff
@@ -278,20 +314,22 @@ export default Ember.Controller.extend(CanCheckEmails, {
       this.get("model").generateApiKey();
     },
 
-    groupAdded(added) {
-      this.get("model")
-        .groupAdded(added)
-        .catch(function() {
-          bootbox.alert(I18n.t("generic_error"));
-        });
+    saveCustomGroups() {
+      const currentIds = this.get("customGroupIds");
+      const bufferedIds = this.get("customGroupIdsBuffer");
+      const availableGroups = this.get("availableGroups");
+
+      bufferedIds.filter(id => !currentIds.includes(id)).forEach(id => {
+        this.groupAdded(availableGroups.findBy("id", id));
+      });
+
+      currentIds
+        .filter(id => !bufferedIds.includes(id))
+        .forEach(id => this.groupRemoved(id));
```

Comment:
Note, even though this is issuing the `POST`/`DELETE` requests one by one, I don't think it warrants a new endpoint.